diff --git a/single_image.module b/single_image.module
index e0af4ff..b4a7743 100644
--- a/single_image.module
+++ b/single_image.module
@@ -62,6 +62,9 @@ function single_image_field_formatter_view($entity_type, $entity, $field, $insta
   $element = array();
   if ($items) {
     $delta = $display['settings']['single_image_delta'];
+    if($delta > 0) {
+      $delta = $delta - 1;
+    }
     $which = isset($items[$delta]) ? $delta : array_rand($items);
     $items = array($items[$which]);
     $element = image_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display);
